sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/specialized/ShareLeaseAsyncClient.java

```diff
@@ -246,9 +324,62 @@ public String getLeaseId() {
 
     Mono<Response<String>> changeLeaseWithResponse(String proposedId, Context context) {
         context = context == null ? Context.NONE : context;
-        return this.client.files().changeLeaseWithRestResponseAsync(null, null, this.leaseId, null, proposedId,
-            null, context.addData(AZ_TRACING_NAMESPACE_KEY, STORAGE_TRACING_NAMESPACE_VALUE))
-            .map(rb -> new SimpleResponse<>(rb, rb.getDeserializedHeaders().getLeaseId()));
+        if (this.isShareFile) {
+            return this.client.files().changeLeaseWithRestResponseAsync(null, null, this.leaseId, null, proposedId,
+                null, context.addData(AZ_TRACING_NAMESPACE_KEY, STORAGE_TRACING_NAMESPACE_VALUE))
+                .map(rb -> new SimpleResponse<>(rb, rb.getDeserializedHeaders().getLeaseId()));
+        } else {
+            return this.client.shares().changeLeaseWithRestResponseAsync(null, this.leaseId, null, proposedId, null,
+                null, context.addData(AZ_TRACING_NAMESPACE_KEY, STORAGE_TRACING_NAMESPACE_VALUE))
+                .map(rb -> new SimpleResponse<>(rb, rb.getDeserializedHeaders().getLeaseId()));
+        }
+    }
+
+    /**
+     * Renews the previously acquired lease on a share.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.specialized.ShareLeaseAsyncClient.renewLease}
+     *
+     * @return A reactive response containing the renewed lease ID.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<String> renewLease() {
+        try {
+            return renewLeaseWithResponse().flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Renews the previously acquired lease on a share.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.specialized.ShareLeaseAsyncClient.renewLeaseWithResponse}
+     *
+     * @return A reactive response containing the renewed lease ID.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<String>> renewLeaseWithResponse() {
+        try {
+            return withContext(this::renewLeaseWithResponse);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<String>> renewLeaseWithResponse(Context context) {
+        context = context == null ? Context.NONE : context;
+        if (this.isShareFile) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("Cannot renew a lease on a share file."));
```

Comment:
Should this be an UnsupportedOperationException?

Reply:
Yeah youre right. I will change this